airvantage.go

```diff
@@ -101,8 +105,69 @@ type apiError struct {
 func (av *AirVantage) parseResponse(resp *http.Response, respStruct interface{}) error {
 	defer resp.Body.Close()
 
+	if err := av.parseError(resp); err != nil {
+		return err
+	}
+
+	if respStruct == nil {
+		return nil
+	}
+
+	var payload io.Reader = resp.Body
+	if av.Debug {
+		body, err := io.ReadAll(resp.Body)
+		if err != nil {
+			return err
+		}
```

Comment:
When I see this construct is duplicated all along go files, I really like C macros